build/build.sh

```diff
@@ -0,0 +1,187 @@
+    echo
+    exit 1
+}
+
+# Parse arguments
+while (("$#")); do
+    case "$1" in
+    -a | --deepdetect-arch)
+        DEEPDETECT_ARCH=$2
+        shift 2
+        ;;
+    -b | --deepdetect-build)
+        DEEPDETECT_BUILD=$2
+        shift 2
+        ;;
+    -h | --help)
+        help_menu
+        break
+        ;;
+    esac
+done
+
+# Deepdetect platform selector
+select_platform() {
+    clear
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo " Select Deepdetect build platform"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    for index in ${!deepdetect_arch[*]}; do
+        echo "  $index. ${deepdetect_arch[$index]^^}"
+    done
+    echo ""
+
+    local choice
+    read -p "Enter choice : " choice
+    DEEPDETECT_ARCH=${deepdetect_arch[$choice]}
+}
+
+# Deepdetect select CPU build profile
+select_cpu_build_profile() {
+    clear
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo " Select Deepdetect build profile"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    for index in ${!deepdetect_cpu_build_profiles[*]}; do
+        echo "  $index. ${deepdetect_cpu_build_profiles[$index]}"
+    done
+    echo ""
+
+    local choice
+    read -p "Enter choice : " choice
+    DEEPDETECT_BUILD=${deepdetect_cpu_build_profiles[$choice]}
+}
+
+# Deepdetect select GPU build profile
+select_gpu_build_profile() {
+    clear
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo " Select Deepdetect build profile"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    for index in ${!deepdetect_gpu_build_profiles[*]}; do
+        echo "  $index. ${deepdetect_gpu_build_profiles[$index]}"
+    done
+    echo ""
+    
+    local choice
+    read -p "Enter choice : " choice
+    DEEPDETECT_BUILD=${deepdetect_gpu_build_profiles[$choice]}
+}
+
+# Use menu...
+show_interactive_platform_selector() {
+    # Main menu handler loop
+    select_platform
+    if [ ${DEEPDETECT_ARCH} == "cpu" ]; then
+        select_cpu_build_profile
+    fi
+    if [ ${DEEPDETECT_ARCH} == "gpu" ]; then
+        select_gpu_build_profile
+    fi
+}
+
+# Build functions
+cpu_build() {
+
+    case ${DEEPDETECT_BUILD} in
+
+    "caffe-tf")
+        cmake .. -DUSE_TF=ON -DUSE_TF_CPU_ONLY=ON -DUSE_SIMSEARCH=ON -DUSE_TSNE=ON -DUSE_NCNN=OFF
+        make -j
+        ;;
+
+    "armv7")
+        cmake .. -DUSE_NCNN=ON -DRPI3=ON -DUSE_HDF5=OFF -DUSE_CAFFE=OFF
+        make -j
+        ;;
+
+    *)
+        cmake .. -DUSE_XGBOOST=ON -DUSE_SIMSEARCH=ON -DUSE_TSNE=ON -DUSE_NCNN=ON
+        make -j
+        ;;
+    esac
+
+}
+
+gpu_build() {
+
+    case ${DEEPDETECT_BUILD} in
+
+    "caffe-cpu-tf")
+        cmake .. -DUSE_TF=ON -DUSE_TF_CPU_ONLY=ON -DUSE_CUDNN=ON -DUSE_XGBOOST=ON -DUSE_SIMSEARCH=ON -DUSE_TSNE=ON -DCUDA_ARCH="-gencode arch=compute_30,code=sm_30 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_53,code=sm_53 -gencode arch=compute_61,code=sm_61"
```

Comment:
@fantes Could you review `-DCUDA_ARCH` options in multiple `DEEPDETECT_BUILD` targets ?

Thanks, Alex

Reply:
https://docs.nvidia.com/cuda/turing-compatibility-guide/index.html

Reply:
> 30 et 35 concernent des vieilles vieilles cartes plus supportées je pense , et 53 et 62 n'ont jamais existé je pense

Flags to use depending on chosen `CUDA_VERSION`:

* Cuda 9

```
  -gencode=arch=compute_50,code=sm_50
  -gencode=arch=compute_52,code=sm_52
  -gencode=arch=compute_60,code=sm_60
  -gencode=arch=compute_61,code=sm_61
  -gencode=arch=compute_70,code=sm_70
  -gencode=arch=compute_70,code=compute_70
```

* Cuda 10

```
  -gencode=arch=compute_50,code=sm_50
  -gencode=arch=compute_52,code=sm_52
  -gencode=arch=compute_60,code=sm_60
  -gencode=arch=compute_61,code=sm_61
  -gencode=arch=compute_70,code=sm_70
  -gencode=arch=compute_75,code=sm_75
  -gencode=arch=compute_75,code=compute_75
```